diff --git a/nodepool/driver/openshiftpods/provider.py b/nodepool/driver/openshiftpods/provider.py
index d1e05f6..11f2e69 100644
--- a/nodepool/driver/openshiftpods/provider.py
+++ b/nodepool/driver/openshiftpods/provider.py
@@ -119,14 +119,8 @@
             self.log.exception("%s: unknown pool" % server_id)
             return
         self.log.debug("%s: removing pod" % pod_name)
-        delete_body = {
-            "apiVersion": "v1",
-            "kind": "DeleteOptions",
-            "propagationPolicy": "Background"
-        }
         try:
-            self.k8s_client.delete_namespaced_pod(
-                pod_name, project_name, delete_body)
+            self.k8s_client.delete_namespaced_pod(pod_name, project_name)
             self.log.info("%s: pod removed" % server_id)
         except Exception:
             # TODO: implement better exception handling
